docs/advanced/fields.rst

```diff
@@ -314,63 +292,63 @@ to the item:
 
 .. code-block:: python
 
+    import attrs
+    from my_library import FooPage, StandardItem
+    from web_poet import HttpResponse, field, Returns
+
     @attrs.define
     class CustomItem(StandardItem):
         new_field: str
 
     @attrs.define
-    class CustomFooPage(FooPage):
+    class CustomFooPage(FooPage, Returns[CustomItem]):
 
         @field
         def new_field(self) -> str:
             # ...
 
-        async def to_item(self) -> CustomItem:
-            # we need to override to_item to ensure CustomItem is returned
-            return await item_from_fields(self, item_cls=CustomItem)
+Note how :class:`~.Returns` is used as one of the base classes of
+``CustomFooPage``; it allows to change the item type returned by a page object.
 
-Removing fields (as well as renaming) is more tricky with inheritance though.
+Removing fields (as well as renaming) is a bit more tricky.
 
-The caveat is that by default :func:`item_from_fields` uses all fields
+The caveat is that by default :class:`~.ItemPage` uses all fields
 defined as ``@field`` to produce an item, passing all these values to
-``Item.__init__``. So, if you follow the previous example, and inherit from
-the "base", "standard" Page Object, there could be a ``@field`` from the base
-class which is not present in the ``CustomItem``. It'd be still passed
-to ``CustomItem.__init__``, causing an exception.
+item's ``__init__`` method. So, if you follow the previous example, and
+inherit from the "base", "standard" Page Object, there could be a ``@field``
+from the base class which is not present in the ``CustomItem``.
+It'd be still passed to ``CustomItem.__init__``, causing an exception.
 
-To solve it, you can either
+One way to solve it is to make the orignal Page Object a dependency
+instead of inheriting from it, as explained in the beginning.
 
-* make the orignal Page Object a dependency instead of inheriting from it
-  (as explained in the beginning), or
-* use ``item_cls_fields=True`` argument of :func:`item_from_fields`:
-  when ``item_cls_fields`` parameter is True, ``@fields`` which
-  are not defined in the item are skipped.
+Alternatively, you can use ``skip_nonitem_fields=True`` class argument - it tells
```

Comment:
I'm on fence about this suggestion :) On one hand, it makes sense. On the other hand, class arguments are so rarely used, so that thinking about default values of class arguments could be too much. I think for most users it'd be just "copy-paste ``skip_nonitem_fields=True``". No strong opinon on it though.